### ISSUE TITLE
Flag Task.Delay(0) as being Task.CompletedTask

### ISSUE THREAD

**Issue by Keboo** [IntelliTect/CodingStandards#55]
Task.Delay(0) is the same as Task.CompletedTask
[.NET Core](https://source.dot.net/#System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs,5396)
[.NET Framework](https://referencesource.microsoft.com/#mscorlib/system/threading/Tasks/Task.cs,5881)

Test cases:
```
Task.Delay(0);
Task.Delay(0, token);
```

Suggest:
We should suggest changing to `Task.CompletedTask` since that is what is going to happen anyway.

**Comment by adamskt** [comment]
@ascott18  Is there a Roslynator analyzer that catches this?

**Comment by ascott18** [comment]
@adamskt No, there isn't. This seems like such an unlikely thing for someone to do that I can't imagine its worth any effort to build such an analyzer ourselves, either.
